### PR TITLE
Add sidebar as an option

### PR DESCRIPTION
**Pull request by seqizz**
Well I wouldn't think I can manage to run that, but seems like working.

Please check, since I probably broke the .ts file.

### CODE REVIEW

**Comment by tomasbonco** [comment]
Sorry for my late reply, I'm very tight on schedule this days. I only had a few moments today to test this. First of all I think idea is really good. The actual code need a few more touches:
* inject.js and inject.ts do not match. That is weird as inject.js should be just generated
* `browser.storage.local.get...`  breaks the extension as browser is `undefined`
* option.js should be generated from option.ts as well (to keep consistency)
and a few more ...

So basically I really really really appreciate your idea and your work. Hopefully next week I'll try to extend the extension based on it (or if you have time you keep commiting :D). Thank you very much for making this extension better.

**Comment by seqizz** [comment]
Yeah, thanks :smile: I am not a developer and sent MR as soon as I ran it successfully on my Firefox. (.ts format just seems alien to me) :fearful: 
If you want to test how that would feel like on sidebar mode, you can try [this](https://addons.mozilla.org/en-US/firefox/addon/gitlab-tree-view-sidebar/) Firefox extension. I'll be deleting that one once you give something similar. That was a fun ride poking javascript but unfortunately my mind is not capable of such weird language.

**Comment by tomasbonco** [comment]
I decided not to add the extension to the sidebar. There are a few reasons:
1. Usually, the sidebar is closed.
2. It doesn't match the concept and visual of extension. I would like to avoid having two designs.

But I was finding ways how to reduce its size, without adding it to the sidebar. Let me know what you think about it.

Lastly, I was looking one more time at your MR, and I think you did an amazing job. Doing a whole options page and stuff surely required a lot of time and reading of docs. When I was trying to do that I run into a problem that it doesn't work cross-browser (at least not for me), that's why I implemented settings (options) the way I did.

Anyway, thank you very much. I'm closing this one, but I would like to hear your feedback :)

**Comment by seqizz** [comment]
Hi, thanks for evaluation and nice words, but I _know_ my MR was really bad code-wise :)

But I am still using my horrible version and have my personal answers to your two points:

- I am always keeping sidebar open, so that is subjective per user. But I understand if you always use collapsed sidebar, you'll hardly see this as a problem.
- You're "having two designs" with everything which you control via options (or even features like #8 ).

If I have more knowledge or time about this, I'd always just replace the filename list dynamically inside the sidebar whenever it is not collapsed (no option or anything, just as the default behavior). My version doesn't work if the sidebar is collapsed.

Reducing its size is also a way, you can even make that filename list collapsible, but that would be very weird since we already have something on further left which is itself collapsible: The sidebar :trollface: 

But since I am using my fork and that is totally your decision, I'll stop whining :) Thanks for the effort!
